src/ImageLoaders/IHex32/IntelHex32Reader.cs

```diff
@@ -0,0 +1,198 @@
+using Reko.Core;
+
+namespace Reko.ImageLoaders.IntelHex32
+{
+
+    /// <summary>
+    /// A reader capable of reading a Intel Hexadecimal 32-bit format object (a.k.a. IHEX32) stream.
+    /// </summary>
+    public class IntelHex32Reader : IDisposable
+    {
+
+        #region Locals
+
+        private readonly StreamReader streamReader;
+        private int lineNum;
+        private bool IsSegmented = false;
+        private bool IsLinear = false;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructs instance of an <see cref="IntelHex32Reader" />.
+        /// </summary>
+        /// <param name="str">The source stream of the hex file.</param>
+        /// <exception cref="ArgumentNullException">If the <paramref name="str" /> is null.</exception>
+        public IntelHex32Reader(Stream str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+            streamReader = new StreamReader(str);
+            lineNum = 0;
+        }
+
+        #endregion
+
+        #region IDisposable implementation
+
+        private bool disposedValue; // To detect redundant calls
+
+        /// <summary>
+        /// Dispose the <see cref="IntelHex32Reader"/>
+        /// </summary>
+        /// <param name="disposing"></param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    streamReader?.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        /// <summary>
+        /// Dispose the <see cref="IntelHex32Reader"/>
+        /// </summary>
+        public void Dispose()
+        {
+            // Do not change this code. Put cleanup code in Dispose(disposing) above.
+            Dispose(true);
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Gets or sets the base address for loading/dumping the 16-bit IHex32 records.
+        /// </summary>
+        public uint AddressBase { get; set; } = 0;
+
+        /// <summary>
+        /// Gets the start address (program entry point).
+        /// </summary>
+        public Address StartAddress { get; set; } = null;
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Read data with address information from the stream.
+        /// </summary>
+        /// <param name="address">[out] The start address for the block of data.</param>
+        /// <param name="data">[out] The data byte block.</param>
+        /// <returns>
+        /// true if there are more records available or false if end of file.
+        /// </returns>
+        /// <remarks>
+        /// An address value may be read without corresponding data bytes. This occurs for record
+        /// types 02, 04, 05.
+        /// </remarks>
+        public bool Read(out uint address, out byte[] data)
```

Comment:
The naming pattern we follow here is established in .NET. If you are return values in out parameters, consider renaming to `TryRead` (see, for instance `Dictionary<K,V>.TryGetValue(..)`).

And what exactly are we reading? Perhaps `TryReadHexRecord` is more explanatory?

Reply:
Thanks for this comment. Corrected.